### ISSUE TITLE
Add Google mock framework as a dependency so we can use it for mocking in tests.

### ISSUE THREAD

**Issue by schmie** [schmie/hemeMove#103]
**Reported by jamespjh on 21 Nov 2011 10:12 UTC**
This would be a testing dependency only (same as cppunit) so wouldn't need to work on production targets.

http://code.google.com/p/googlemock/wiki/ForDummies

**Comment by schmie** [comment]
**Comment by hywel on 21 Nov 2011 10:55 UTC**
Yup, fine with me. It'd be nice to have a wiki page about how to use it well (a condensed version of the wiki on Google Code would do).

**Comment by schmie** [comment]
**Comment by miguel on 21 Nov 2011 11:40 UTC**
Sounds reasonable to me. James, how does this change what we discussed the other day re #23 (mocking the phased broadcast class, etc.)?

**Comment by schmie** [comment]
**Comment by miguel on 24 Nov 2011 11:49 UTC**
Blocked on dependency server (#44). Moving to 0.3.

**Comment by schmie** [comment]
**Modified by miguel on 24 Nov 2011 11:49 UTC**

**Comment by schmie** [comment]
**Modified by jamespjh on 24 Nov 2011 15:01 UTC**

**Comment by schmie** [comment]
**Modified by hywel on 1 Mar 2012 17:37 UTC**

**Comment by schmie** [comment]
**Modified by jamespjh on 7 Mar 2012 19:45 UTC**

**Comment by schmie** [comment]
**Modified by hywel on 8 Mar 2012 14:10 UTC**

**Comment by schmie** [comment]
**Modified by jamespjh on 5 Jul 2012 15:49 UTC**
